extensions/markdown-language-features/src/lines.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+
+
+export function createTextLine(line: number, offset: number, text: string, lineBreak: string): vscode.TextLine {
+	let wsRe = /s*/;
+	let firstNws = 0;
+	if (wsRe.test(text)) {
+		firstNws = wsRe.lastIndex;
+	}
+	return {
+		lineNumber: line,
+		text: text,
+		range: new vscode.Range(line, offset, line, offset + text.length),
+		rangeIncludingLineBreak: new vscode.Range(line, offset, line, offset + text.length + lineBreak.length),
```

Comment:
Do we really need to create a ` vscode.TextLine` or can we create a more minimal interface for a line that only includes the properties we really need?

Reply:
We don't but it was suggested by @OmarTawfik and I think reusing existing interfaces is a good idea in cases where we will need to reuse this code in the future.

Reply:
I introduced SkinnyTextLine which only contains text. This substantially simplified code, so that I reduced a number of files.